packages/build/src/steps/error.ts

```diff
@@ -81,35 +86,51 @@ const handleFailPlugin = async function ({
   logs,
   debug,
   testOpts,
-}) {
+}: failPluginArgs) {
   const newStatus = serializeErrorStatus({ fullErrorInfo, state: 'failed_plugin' })
   await handleBuildError(newError, { errorMonitor, netlifyConfig, childEnv, mode, logs, debug, testOpts })
-  return { failedPlugin: [packageName], newStatus }
+  // TODO we should probably use type guard here, but due to the way we build these errorInfo objects I'm not 100%
+  // confident we have all the properties currently required by the type
+  const location = fullErrorInfo.errorInfo.location as PluginLocation
+  return { failedPlugin: [location.packageName], newStatus }
 }
 
-// On `utils.build.cancelBuild()`
+/* On `utils.build.cancelBuild()` */
 const handleCancelBuild = async function ({ fullErrorInfo, newError, api, deployId }) {
   const newStatus = serializeErrorStatus({ fullErrorInfo, state: 'canceled_build' })
   await cancelBuild({ api, deployId })
   return { newError, newStatus }
 }
 
-// On `utils.build.failBuild()` or uncaught exception
+/* On `utils.build.failBuild()` or uncaught exception */
 const handleFailBuild = function ({ fullErrorInfo, newError }) {
   const newStatus = serializeErrorStatus({ fullErrorInfo, state: 'failed_build' })
   return { newError, newStatus }
 }
 
-// Unlike community plugins, core plugin bugs should be handled as system errors
-export const getPluginErrorType = function (error, loadedFrom) {
+/* Unlike community plugins, core plugin and trusted plugin bugs should be handled as system errors */
+export const getPluginErrorType = function (
+  error: Error,
+  loadedFrom: string,
+  packageName?: string,
+): { type?: ErrorTypes } {
+  console.error(error)
```

Comment:
Did you mean to leave this here?

Reply:
🤦 definitely not, thanks for spotting it. Fixed via - https://github.com/netlify/build/pull/5408/commits/cd7a62ce02e9bbf7e275283a49694f5ef4d9ac73